tests/unit/test_query_language.py

```diff
@@ -63,3 +67,45 @@ def test_dataset() -> None:
             rhs=Value(2000),
         ),
     }
+
+
+# The problem: We'd like to test that operations on query language (QL) elements return
+# the correct query model (QM) elements. We like tests that emphasise what is being
+# tested, and de-emphasise the scaffolding. We dislike test code that looks like
+# production code.
+@pytest.fixture
+def select_patient_table():
+    return SelectPatientTable(name="patient_table")
+
+
+@pytest.fixture
+def select_column(select_patient_table):
+    return SelectColumn(select_patient_table, name="column")
+
+
+qm = operator.attrgetter("qm_node")
+
+
+def assert_produces(ql_element, qm_element):
+    assert qm(ql_element) == qm_element
+
+
+class TestIntSeries:
+    def test_le_value(self, select_column):
+        assert_produces(
+            IntSeries(select_column) <= 2000, Function.LE(select_column, Value(2000))
+        )
```

Comment:
It would be nice if `select_column` were called something far less specific. I think it could actually be any QM series object, so some name that reflects that would be ideal.

Reply:
Thanks for today's discussion about this. 2bd5724 responds to your comment about `select_column` by introducing `qm_select_int` and `qm_select_date` (as variables rather than fixtures). I spent a bit of time thinking about type checking; specifically, why...

```python
Function.LE(
    lhs=SelectColumn(SelectPatientTable(name="some_table"), name="some_column"),
    rhs=Value(2000),
)
```

...didn't raise an error. I then realized this was because `lhs` was `Series[Any]`. So, I've added a `TableSchema` to type the columns.

Reply:
Oh, that's great. I'd missed the need for a schema. And adding that schema makes it all the more obvious that putting the types in those variables names is correct.

Reply:
I'm not entirely sure about "select" in that name. I think the series-ness is more interesting to someone reading the code than the fact that it's created by a `SelectColumn`. In a real use of `<=`, it could equally be `AggregateByPatient.Count(SelectPatientTable("events"))`, where we don't even to a `SelectColumn`.

Reply:
Thanks for your patience -- we discussed this, and then I forgot we discussed this 🤦🏻‍♂️. 9ab9d0a emphasizes the series-ness over the select-ness and also de-emphasizes the patient-ness of the table and schema.